storelogs/main.go

```diff
@@ -0,0 +1,117 @@
+// pmm-agent
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package storelogs help to store logs
+package storelogs
+
+import (
+	"bytes"
+	"container/ring"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// LogsStore implement ring save logs
+type LogsStore struct {
+	log   *ring.Ring
+	Entry *logrus.Entry
+	count int
+	m     sync.RWMutex
+}
+
+// Init initializes basic parameters
+func (l *LogsStore) Init(entry *logrus.Entry) {
+	if l.count == 0 {
+		l.count = 10
+	}
+	l.log = ring.New(l.count)
+	l.Entry = entry
+}
+
+// SetCountLogs sets the number of logs to store
+func (l *LogsStore) SetCountLogs(countLogs int) {
+	l.count = countLogs
+}
+
+func (l *LogsStore) saveLog(log string) {
```

Comment:
probably we could do it using Hooks mechanism or by replacing output in logrus, so we won't be need to do that.

Reply:
> probably we could do it using Hooks mechanism or by replacing output in logrus, so we won't be need to do that.


this is not an exported function and they work with standard logrus functions, I didn’t manage to add this when outputting, so I made wrapper functions like Infof, Debugf etc..

Reply:
I prepared PR with changes I suggested https://github.com/percona/pmm-agent/pull/364, Please take a look